extern/boostd-data/svc/svc_test.go

```diff
@@ -340,14 +344,131 @@ func compareIndices(subject, subjectDb index.Index) (bool, error) {
 	var b bytes.Buffer
 	w := bufio.NewWriter(&b)
 
-	subject.Marshal(w)
+	_, err := subject.Marshal(w)
+	if err != nil {
+		return false, err
+	}
 
 	var b2 bytes.Buffer
 	w2 := bufio.NewWriter(&b2)
 
-	subjectDb.Marshal(w2)
+	_, err = subjectDb.Marshal(w2)
+	if err != nil {
+		return false, err
+	}
 
 	res := bytes.Compare(b.Bytes(), b2.Bytes())
 
 	return res == 0, nil
 }
+
+func TestCleanup(t *testing.T) {
+	_ = logging.SetLogLevel("*", "debug")
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+	defer cancel()
+
+	t.Run("level db", func(t *testing.T) {
+		bdsvc, err := NewLevelDB("")
+		require.NoError(t, err)
+		testCleanup(ctx, t, bdsvc, 8042)
+	})
+	t.Run("couchbase", func(t *testing.T) {
+		removeContainer := setupCouchbase(t)
+		defer removeContainer()
+		bdsvc := NewCouchbase(testCouchSettings)
+		testCleanup(ctx, t, bdsvc, 8043)
+	})
+}
+
+func testCleanup(ctx context.Context, t *testing.T, bdsvc *Service, port int) {
+	err := bdsvc.Start(ctx, port)
+	require.NoError(t, err)
+
+	cl := client.NewStore()
+	err = cl.Dial(context.Background(), "http://localhost:8042")
+	require.NoError(t, err)
+	defer cl.Close(ctx)
+
+	sampleidx := "fixtures/baga6ea4seaqnfhocd544oidrgsss2ahoaomvxuaqxfmlsizljtzsuivjl5hamka.full.idx"
+
+	pieceCid, err := cid.Parse("baga6ea4seaqnfhocd544oidrgsss2ahoaomvxuaqxfmlsizljtzsuivjl5hamka")
+	require.NoError(t, err)
+
+	subject, err := loadIndex(sampleidx)
+	require.NoError(t, err)
+
+	records, err := getRecords(subject)
+	require.NoError(t, err)
+
+	randomuuid, err := uuid.Parse("4d8f5ce6-dbfd-40dc-8b03-29308e97357b")
+	require.NoError(t, err)
+
+	err = cl.AddIndex(ctx, pieceCid, records)
+	require.NoError(t, err)
+
+	di := model.DealInfo{
+		DealUuid:    randomuuid,
+		SectorID:    abi.SectorNumber(1),
+		PieceOffset: 1,
+		PieceLength: 2,
+		CarLength:   3,
+	}
+
+	// Add a deal for the piece
+	err = cl.AddDealForPiece(ctx, pieceCid, di)
+	require.NoError(t, err)
+
+	// Add the same deal a second time to test uniqueness
+	err = cl.AddDealForPiece(ctx, pieceCid, di)
+	require.NoError(t, err)
```

Comment:
This is already tested elsewhere, let's try to keep this test to the minimum required to test the remove functionality